arches_references/src/arches-references/utils.ts

```diff
@@ -40,22 +40,32 @@ export const languageNameFromCode = (code: string) => {
     return arches.languages.find((lang: Language) => lang.code === code).name;
 };
 
-export const findNodeInTree = (tree: TreeNode[], itemId: string) => {
+export const findNodeInTree = (
+    tree: TreeNode[],
+    itemId: string,
+): {
+    found: TreeNode | undefined;
+    path: TreeNode[];
+} => {
+    const path: TreeNode[] = [];
+
```

Comment:
This refactor adds the ability to do a lookup in a tree by item id, which wasn't something was needed before (the user was clicking on the tree to navigate to items). We need this for routing to be able to go from a path to somewhere arbitrary in the tree.